src/components/side-nav/components/SideNavItem.jsx

```diff
@@ -1,13 +1,35 @@
 import React from "react";
 import { Link } from "react-router-dom"; 
-/*is an element that lets the user navigate to another page by clicking or tapping on it.
+/* is an element that lets the user navigate to another page by clicking or tapping on it.
 In react-router-dom, a <Link> renders an accessible <a> element with a real href that points 
 to the resource it's linking to. */
 
-const SideNavItem = ({ title, icon, route }) => {
+/* This hook returns the current 'location' object. This can be useful if
+you'd like to perform some side effect whenver the current location changes. */
+import { useLocation } from "react-router-dom";
+
+const SideNavItem = ({ title, icon, route  }) => {
+    let location = useLocation();
+
+    console.log(location.pathname);
+    
+    
+    /* Adding logic 'IF' and 'ELSE' statements to make sure we set 
+    the correct css classes inside of the component(buidling blocks
+    of any React app). */
+    const getClassName = () => {
+        if (location.pathname === route) {
+            return "flex items-center fill-[#FFFFFF] text-[#FFFFFF] pb-2 px-4 cursor-pointer gap-2.5 hover:text-white hover:fill-white"
+            //returns class names to make text white.
+        } else {
+            return "flex items-center fill-[#b3b3b3] text-[#b3b3b3] pb-2 px-4 cursor-pointer gap-2.5 hover:text-white hover:fill-white"
+            //returns class names to make text grey.
```

Comment:
yessss excellent! this works just fine!

let's take it to the next level and clean this logic up. notice on line 22 we have `hover:text-white hover:fill-white`. those technically are not necessary since the text of the sidenav item will always be white if `location.pathname === route`

let's also clean this up! notice how both line 22 and line 25 contain `flex items-center pb-2 px-4 cursor-pointer gap-2.5`

we could do something like this to make it "cleaner"

```javascript
const baseClasses = `flex items-center pb-2 px-4 cursor-pointer gap-2.5`;

if (location.pathname === route) {
 return `${baseClasses} fill-[#FFFFFF] text-[#FFFFFF]`;
} else {
return `${baseClasses} fill-[#b3b3b3] text-[#b3b3b3] hover:text-white hover:fill-white`;
}
```